make build-image broke after a make clean

**What happened**:
make clean && make build-image is now failing in master:

$ ORG=eriknelson TAG=apb-role-test make build-image
cp broker build/broker
cp: cannot stat 'broker': No such file or directory
make: *** [Makefile:50: build-image] Error 1

**What you expected to happen**:
make build-image should correctly have dep targets setup to fully build images after a clean

--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,7 @@ $(KUBERNETES_FILES):
 prepare-local-env: $(KUBERNETES_FILES) ## Prepare the local environment for running the broker locally
 	@echo > /dev/null
 
-build-image: ## Build a docker image with the broker binary
+build-image: broker | ## Build a docker image with the broker binary
 	cp broker build/broker
 	docker build -f ${BUILD_DIR}/Dockerfile-localdev -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR}
 	@echo

--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,7 @@ $(KUBERNETES_FILES):
 prepare-local-env: $(KUBERNETES_FILES) ## Prepare the local environment for running the broker locally
 	@echo > /dev/null
 
-build-image: broker | ## Build a docker image with the broker binary
+build-image: broker ## Build a docker image with the broker binary
 	cp broker build/broker
 	docker build -f ${BUILD_DIR}/Dockerfile-localdev -t ${BROKER_IMAGE}:${TAG} ${BUILD_DIR}
 	@echo

